rwengine/src/script/ScriptFunctions.hpp

```diff
@@ -0,0 +1,211 @@
+#ifndef RWENGINE_SCRIPTFUNCTIONS_HPP
+#define RWENGINE_SCRIPTFUNCTIONS_HPP
+
+#include <engine/GameWorld.hpp>
+#include <engine/GameState.hpp>
+#include <engine/GameData.hpp>
+#include <objects/GameObject.hpp>
+#include <script/ScriptTypes.hpp>
+#include <script/ScriptMachine.hpp>
+
+/**
+ * Implementaitions for common functions likely to be shared
+ * among many script modules and opcodes.
+ */
+namespace script {
+const ScriptVec3 kSpawnOffset = ScriptVec3(0.f, 0.f, 1.f);
+
+inline void getObjectPosition(GameObject* object,
+                              ScriptFloat& x,
+                              ScriptFloat& y,
+                              ScriptFloat& z)
+{
+	const auto& p = object->getPosition();
+	x = p.x;
+	y = p.y;
+	z = p.z;
+}
+inline void setObjectPosition(GameObject* object, const ScriptVec3& coord)
+{
+	object->setPosition(coord + script::kSpawnOffset);
+}
+
+inline VehicleObject* getCharacterVehicle(CharacterObject* character)
+{
+	return character->getCurrentVehicle();
```

Comment:
Had you done anything else?

I've tried cancelling the mission with a fare and before picking up a fare and not crashed.